### ISSUE TITLE
Windows Local Research Python Site-Packages

### ISSUE THREAD

**Issue by C-SELLERS** [QuantConnect/Lean#5659]
#### Expected Behavior

Local python research environment works out of box with our start.py script.

#### Actual Behavior

It appears that in some windows environments using Lean research that python does not use the primary site-packages directory on its path by default. Its unclear whether or not this is PythonNet or Python, but the issue does not occur outside of Lean or Lean research for me. Such as using Python in powershell, or just creating a test py file and running  it directly the site-packages is on path path by default.

example path just starting Research
```
['C:\\Users\\Colton\\Desktop\\Working\\QuantConnect\\Repos\\Lean\\Launcher\\bin\\Debug', 'c:\\dev\\python368\\python36.zip', 'c:\\dev\\python368\\DLLs', 'c:\\dev\\python368\\lib', 'c:\\dev\\python368', '', 'C:\\Users\\Colton\\AppData\\Roaming\\Python\\Python36\\site-packages', 'c:\\dev\\python368\\lib\\site-packages\\win32', 'c:\\dev\\python368\\lib\\site-packages\\win32\\lib', 'c:\\dev\\python368\\lib\\site-packages\\Pythonwin', 'c:\\dev\\python368\\lib\\site-packages\\IPython\\extensions', 'C:\\Users\\Colton\\.ipython', 'C:\\Program Files\\dotnet\\shared\\Microsoft.NETCore.App\\5.0.3\\']
```

As you can see it has plenty of site-package directories, but not the primary site-package dir.
`'c:\\dev\\python368\\lib\\site-packages'`
Because of this the required modules like `matplotlib`, `pandas`, etc. Can not be found. As a workaround I am able to run the following cell to add this to the path dynamically.

```
import site
site_packages_path = site.getsitepackages()
for site_package_path in site_packages_path:
    if site_package_path not in sys.path:
        sys.path.append(site_package_path)
```        

#### Potential Solution

Best case solution would require that it is something in our configuration causing this to happen (PythonNet?), 

An option that would cover the research case would be to add the above snippet to our `start.py` script. Although this would fix it for research it would not fix it for Lean. (I observe this issue when running lean regressions as well)

#### Reproducing the Problem

This bug is problematic because it doesn't affect all windows machines, in fact I had this issue occurring a few months ago and after a fresh install of Python it went away. Yet as of last week the issue has started again. 

#### System Information

Windows bug for sure.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

**Comment by C-SELLERS** [comment]
After some digging and debugging it appears there is an issue linked to our stubs package. Although its not exactly clear what the issue is, I did have a breakthrough with the following steps.


remove stubs -> pip uninstall quantconnect-stubs
remove lean-cli -> pip uninstall lean
reinstall lean-cli -> pip install lean

**Comment by C-SELLERS** [comment]
Just a little extra information for future reference:
-  The bug would reproduce every single time I used lean while I had the "bugged" stubs. Meaning that I was unable to run Python algorithms or Python research locally without some hacky adjustments.
    -  With Python Algorithms the issue (site-packages being removed from pythons path) always occurred during the AlgorithmPythonWrapper line `var module = Py.Import(moduleName);`, which imports the algorithm.
- Removing the QuantConnect directory directly from my "site-packages" stopped the bug from occurring.
- If I moved all of these packages to another directory on my Python Path, that directory would disappear too. 

This leads me to believe that somewhere behind the scenes, upon import of the algo, it failed to use the modules in a given directory and dropped the directory from the path. I immediately think of the CLR references in our python algorithms and what they might be attempting to import from the stubs and how that could be part of the issue.
